### PR TITLE
Adding analysis for charge correlations for leading particles in jet

### PR DESCRIPTION
**Pull request by mmriganka**
This one for data only now MC particles would be added in later stage.

### CODE REVIEW

File: PWGJE/Tasks/jetChCorr.cxx [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+      fastjetutilities::fillTracks(jetConstituent, jetConstituents, jetConstituent.globalIndex());
+
+      // int tr = jetConstituent.globalIndex();
+      // auto const& trackSel = tracks.iteratorAt(tr);
+
+      TVector3 p_leading(jetConstituent.px(), jetConstituent.py(), jetConstituent.pz());
+
+      if (nn == 0) {
+        // leadpt =  p_leading.Perp();
+        trackL = jetConstituent.globalIndex();
```
Comment (nzardosh) [inline PWGJE/Tasks/jetChCorr.cxx:354]: i do not think its gaurunteed that the jet constituents are returned in a pT ordered way. You should explicitly loop over them and find the highest pT ones

File: PWGJE/Tasks/jetChCorr.cxx [hunk 2]
```diff
@@ -0,0 +1,439 @@
...
+
+      vector<fastjet::PseudoJet> constituents1 = parentSubJet1.constituents();
+      vector<fastjet::PseudoJet> constituents2 = parentSubJet2.constituents();
+      // cout<<"in subJET1 ********************************************* "<<endl;
+      int found1 = 0;
+      int found2 = 0;
+      for (unsigned int j = 0; j < constituents1.size(); j++) {
+        // cout<<constituents1[j].user_index()-1<<", ";
+        if ((n_trackL == (constituents1[j].user_index() - 1)) || (trackL == (constituents1[j].user_index() - 1)))
+          found1++;
```
Comment (nzardosh) [inline PWGJE/Tasks/jetChCorr.cxx:199]: why not use booleans instead?

File: PWGJE/Tasks/jetChCorr.cxx [hunk 3]
```diff
@@ -0,0 +1,439 @@
...
+      for (unsigned int j = 0; j < constituents2.size(); j++) {
+        // cout<<constituents2[j].user_index()-1<<", ";
+        if ((n_trackL == constituents2[j].user_index() - 1) || (trackL == constituents2[j].user_index() - 1))
+          found2++;
+      }
+      // cout<<endl;
+
+      auto z = parentSubJet2.perp() / (parentSubJet1.perp() + parentSubJet2.perp());
+      auto theta = parentSubJet1.delta_R(parentSubJet2);
+      energyMotherVec.push_back(daughterSubJet.e());
```
Comment (nzardosh) [inline PWGJE/Tasks/jetChCorr.cxx:212]: if you only work with histograms and not trees you can remove this and terminate the clustering at the step where the leading particles were found in opposite subjets

**Review by nzardosh — COMMENTED** [review]
Sorry for the delay in getting round to look at this! To me it seems that the analysis is saving alot of things it might not need so for CPU reasons it might be better to really save what is needed. The output tables should also be changed since they are specific to another analysis task and might change there.
Thanks,
Nima

File: PWGJE/Tasks/jetChCorr.cxx (hunk 3 above)
Comment (mmriganka) [reply PWGJE/Tasks/jetChCorr.cxx:212]: terminated.

File: PWGJE/Tasks/jetChCorr.cxx (hunk 2 above)
Comment (mmriganka) [reply PWGJE/Tasks/jetChCorr.cxx:199]: changed to booleans

File: PWGJE/Tasks/jetChCorr.cxx (hunk 1 above)
Comment (mmriganka) [reply PWGJE/Tasks/jetChCorr.cxx:354]: ordered included

File: PWGJE/Tasks/jetChCorr.cxx [hunk 4]
```diff
@@ -0,0 +1,439 @@
...
+      for (unsigned int j = 0; j < constituents1.size(); j++) {
+        // cout<<constituents1[j].user_index()-1<<", ";
+        if ((n_trackL == (constituents1[j].user_index() - 1)) || (trackL == (constituents1[j].user_index() - 1)))
+          found1++;
+      }
+      // cout<<endl;
+      // cout<<"in subJET2 ********************************************* "<<endl;
+      for (unsigned int j = 0; j < constituents2.size(); j++) {
+        // cout<<constituents2[j].user_index()-1<<", ";
+        if ((n_trackL == constituents2[j].user_index() - 1) || (trackL == constituents2[j].user_index() - 1))
```
[the remark this answers is not in the document]
Comment (mmriganka) [reply PWGJE/Tasks/jetChCorr.cxx:205]: It seems that the -1 bring the proper match in track index. Earlier I used without -1, however, I got  wrong answer.

File: PWGJE/Tasks/jetChCorr.cxx [hunk 5]
```diff
@@ -0,0 +1,439 @@
...
+              registry.fill(HIST("h2_jet_pt_part_jet_zg_part"), jet.pt(), zg);
+              registry.fill(HIST("h2_jet_pt_part_jet_rg_part"), jet.pt(), rg);
+            }
+            if constexpr (isSubtracted && !isMCP) {
+              registry.fill(HIST("h2_jet_pt_jet_zg_eventwiseconstituentsubtracted"), jet.pt(), zg);
+              registry.fill(HIST("h2_jet_pt_jet_rg_eventwiseconstituentsubtracted"), jet.pt(), rg);
+            }
+            softDropped = true;
+
+            softDropped = true;
```
[the remark this answers is not in the document]
Comment (mmriganka) [reply PWGJE/Tasks/jetChCorr.cxx:236]: corrected.

File: PWGJE/Tasks/jetChCorr.cxx [hunk 6]
```diff
@@ -0,0 +1,439 @@
...
+TVector3 v1, v2, vR; // leading and next to leading vector
+int ch_l = 0;        // charge leading
+int ch_nl = 0;       // charge next to leading
+
+int ch_mult = 0; // charge next to leading
+
+#include "Framework/runDataProcessing.h"
+
+struct JetChCorr {
+  Produces<aod::CJetSSs> jetSubstructureDataTable;
```
[the remark this answers is not in the document]
Comment (mmriganka) [reply PWGJE/Tasks/jetChCorr.cxx:67]: Excluded table for the time being. May be include stable structure in future.

**Pull request closed without merging**